src/metabase/fingerprinting.clj

```diff
@@ -194,24 +202,29 @@
       [col (fingerprint-field field)])))
 
 (defmethod fingerprint (class Table)
-  [table]
+  [opts table]
   (fingerprint-query (metadata/query-values
                       (:db_id table)
-                      {:source-table (:id table)})))
+                      (merge (extract-query-opts opts)
```

Comment:
Hmmm... I think transforming cost into a query-opt works in the short term but locks us into a specific idiom.
I think we ought to have a "get-data" function that takes a cost, and returns the data to a finger-printer. If the data brings in linked tables, etc it all happens in the get-data function and the result of that hopefully will be of a shape that the downstream fingerprinter knows how to deal with.